Tools/Scripts/generate-static-analysis-archive

```diff
@@ -58,30 +101,49 @@ def parse_command_line(args):
     return parser.parse_args(args)
 
 
-def generate_unexpected_results_links(output_root, static_analysis_dir, project_name):
-    results_links = ''
-    for category, category_formatted in {'UnexpectedPasses': 'Unexpected Passes', 'UnexpectedFailures': 'Unexpected Failures'}.items():
-        command = 'find {} -name {}\\* -print'.format(os.path.abspath(os.path.join(static_analysis_dir, project_name)), category)
-        try:
-            result_files = subprocess.check_output(command, shell=True, text=True)
-        except subprocess.CalledProcessError as e:
-            sys.stderr.write(f'{e.output}')
-            sys.stderr.write(f'Could not find results for {project_name}\n')
-            return -1
-
-        unexpected_results_files = result_files.splitlines()
-        checker_links = ''
-        for file_path in unexpected_results_files:
-            file_name = file_path.split('/')[-1].removeprefix(category).removesuffix('.txt')
-            relative_file_path = os.path.relpath(file_path, output_root)
-            with open(os.path.abspath(os.path.join(output_root, file_path))) as f:
-                count = len(f.read().splitlines())
-            if count:
-                checker_links += PROJECT_TEMPLATE.format(project_file_url=relative_file_path, project_name=file_name, project_issue_count=count)
-        if checker_links:
-            results_links += CATEGORY_TEMPLATE.format(category_name=category_formatted, checker_links=checker_links)
-
-    return results_links
+def generate_unexpected_table(output_root, static_analysis_dir, project_name, category):
+    # Get lists of unexpected results per project
+    category_table = ''
+    command = 'find {} -name {}\\* -print'.format(os.path.abspath(os.path.join(static_analysis_dir, project_name)), category)
+    try:
+        result_files = subprocess.check_output(command, shell=True, text=True)
+    except subprocess.CalledProcessError as e:
+        sys.stderr.write(f'{e.output}')
+        sys.stderr.write(f'Could not find results for {project_name}\n')
+        return -1
+    unexpected_results_files = result_files.splitlines()
+
+    # Add links to unexpected lists and counts in a table
```

Comment:
Rather than adding a comment like this, we may want to extract the below code as a helper method like this:
```
table_rows = create_table_for_unexpected_results_per_project(unexpected_results_per_project)
```